### ISSUE TITLE
Better error description

### ISSUE THREAD

**Issue by trunet** [newrelic/terraform-provider-newrelic#807]
### Feature Description
Trying to create a `newrelic_nrql_alert_condition` resource and noticed that some errors just shows `[{“extensions”:{“code”:“BAD_USER_INPUT”},“message”:“Validation Error”}]`.

Would be nice to have a better error display to make it easier to detect and fix.

### Describe Alternatives
`TF_LOG=trace terraform apply`

### Additional context
In one of my errors, I got this below. It's clear there that I'm getting the error due to using "all" instead of "at_least_once".
```
2020-07-29T13:04:33.077+0200 [DEBUG] plugin.terraform-provider-newrelic_v2.3.0: time="2020-07-29T13:04:33+02:00" level=trace msg="request completed" body="{\"data\":{\"alertsNrqlConditionStaticCreate\":null},\"errors\":[{\"downstreamResponse\":[{\"extensions\":{\"code\":\"BAD_USER_INPUT\",\"exception\":{\"validationErrors\":[{\"name\":\"terms[].thresholdOccurrences\",\"reason\":\"must be any if value function is set to sum\"}]},\"validationErrors\":[{\"name\":\"terms[].thresholdOccurrences\",\"reason\":\"must be any if value function is set to sum\"}]},\"locations\":[{\"column\":3,\"line\":2}],\"message\":\"Validation Error\",\"path\":[\"nrqlConditionStaticCreate\"]}],\"id\":65618574,\"locations\":[{\"column\":4,\"line\":3}],\"message\":\"A downstream error occurred.\",\"path\":[\"alertsNrqlConditionStaticCreate\"]}]}" headers="{\"Cache-Control\":[\"max-age=0, private, must-revalidate\"],\"Content-Security-Policy\":[\"frame-ancestors *.newrelic.com\"],\"Content-Type\":[\"application/json; charset=utf-8\"],\"Date\":[\"Wed, 29 Jul 2020 11:04:32 GMT\"],\"Proxied-By\":[\"Service Gateway\"],\"Served-By\":[\"nerd-graph\"],\"Server\":[\"Cowboy\"],\"Strict-Transport-Security\":[\"max-age=31536000; includeSubDomains\"],\"Vary\":[\"accept-encoding\"]}" method=POST status_code=200 url="https://api.newrelic.com/graphql"
```

**Comment by sanderblue** [comment]
Hi @trunet, thanks for reporting this. We try to bubble up error messages from the API as best we can, but totally agree that there can be some less-than-helpful error messages depending on the scenario. Can you share you config (sensitive data redacted)?

We'll see if there's something we can do on our end to improve on this.

**Comment by trunet** [comment]
My resource definition is inside a module with variables, it's difficult to trace all of them.

This is the definition:
```hcl
resource "newrelic_nrql_alert_condition" "nrql_alert_condition" {
  count = var.alert_type == "nrql" ? 1 : 0

  policy_id   = var.alert_policy_id == "-1" ? newrelic_alert_policy.alert_policy[0].id : var.alert_policy_id
  name        = "${var.alert_division}-${var.alert_tower}-${var.alert_productname}-${var.alert_environment}-${var.alert_condition_name}"
  runbook_url = var.alert_runbook_url
  enabled     = var.alert_condition_enable

  critical {
    threshold_duration    = var.nrql_alert_condition_duration
    operator              = var.alert_condition_operator
    threshold             = var.alert_condition_threshold
    threshold_occurrences = var.alert_condition_time_function
  }

  nrql {
    query       = var.nrql_alert_condition_query
    since_value = var.nrql_alert_condition_since_value
  }

  value_function       = var.nrql_alert_condition_value_function
  violation_time_limit = var.nrql_alert_violation_time_limit
}
```

In my case the problem was`var.nrql_alert_condition_value_function = 'sum'` and `var.alert_condition_time_function = 'ALL'`. After changing `var.alert_condition_time_function = 'AT_LEAST_ONCE'` it worked.

**Comment by ctrombley** [comment]
Resolved by newrelic/newrelic-client-go#408

It will be released later today in v2.4.2.
